feat(workflow): Add pull request template and workflows to enforce new PR title rules

sihyung-maxim opened the pull request:
This PR adds guidelines and examples on titling pull requests within the MSDK repo that Jacob, Jake, and I came up with.

Sort of a brain dump at the moment. Needs more revising. Any additions and suggestions are welcome.

For the bug-related PRs, we could create a workflow that will automatically create a GitHub issue if the PR is resolving a Jira ticket.

Jake-Carter commented:
@sihyung-maxim @Jacob-Scheiffler I simplified the template a bit and moved it to the `.github` folder.  I don't think it's worth overdoing it here on any actions/workflows.  This is a good enough start to avoid empty descriptions/titles

https://github.com/Analog-Devices-MSDK/msdk/pull/553/commits/cf2dca5074ca0f09feb05cb35f9994c7251a0108

sihyung-maxim commented:
> @sihyung-maxim @Jacob-Scheiffler I simplified the template a bit and moved it to the `.github` folder. I don't think it's worth overdoing it here on any actions/workflows. This is a good enough start to avoid empty descriptions/titles
> 
> [cf2dca5](https://github.com/Analog-Devices-MSDK/msdk/commit/cf2dca5074ca0f09feb05cb35f9994c7251a0108)

I haven't pushed in my latest changes yet due to other tasks popping up, but I found a way to standardize our PR titles using conventional commits paired with semantic versioning. I have a workflow that enforces the PR title changes and there are tools that will autogenerate the changelog if we follow this new convention.

I have an email drafted, which describes the changes, that I need to send, so we can decide whether we want to implement this new convention or not. I need to do some more research on the autogeneration of the changelog first.

Jake-Carter commented on .github/workflows/PR_Title_Check.yml:
```diff
@@ -0,0 +1,73 @@
+            build
+            revert
+
+          scopes: |
```
Add an "Other" scope

lorne-maxim replied:
I'm thinking we should add MAXUSB and SDHC also.  Along with BLE and Boards, these libraries are developed in house and not 3rd party.

Jake-Carter commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,98 @@
+
+NOTE: The scope is case sensitive and must match one of the listed scopes.
+
+### Responsibilites for Reviewers/Mergers/PR Owners
```
I suggest moving this into CONTRIBUTING.md.  It makes the PR template too cluttered/long.  Keep in mind that users will need to delete these types of sections from the PR every time

Jake-Carter commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,98 @@
+## Pull Request Template
+
+### Overview
```
Change this section to `### PR Guidelines`, add a template spot for the user's Overview content at the bottom.

Jake-Carter commented on .husky/commit-msg:
What is this for?

sihyung-maxim replied:
Preconfigured git hooks in case you want to install the Commitlint tool locally onto your machine.

lorne-maxim commented on .github/labeler.yml:
```diff
@@ -0,0 +1,172 @@
+- 'Libraries/Include/MAX32690/*'
+
+'MAX78000':
+- 'Examples/MAX78000/*'
```
The MAX78000 uses some of the *me17* files.  (At least it did once upon a time.)

sihyung-maxim replied:
You're right, so does the ME20 -> ME17 and ME16 -> ME15. I've specified the filtering based on the SVD and {part}_files.mk file.

lorne-maxim commented on .github/workflows/PR_Title_Check.yml:
```diff
@@ -0,0 +1,74 @@
+            ignore
+            workflow
+            Other
+
```
If these terms are going to end up in the release notes, I'd like to use "ThirdParty", "Ignore", and "Workflow" to maintain consistent capitalization and punctuation.

lorne-maxim replied:
To clarify, I'm suggesting removing the '-' from "Third-Party", and capitalizing the first letter of "ignore" and "workflow".

lorne-maxim reviewed (approved):
I offered a few minor tweaks that you can implement if you agree, but you can also let it stand as-is.
